api/v1alpha1/project_config_types.go

```diff
@@ -38,6 +46,35 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// PromotionPolicySelector is a selector that matches the resource to which
+// this policy applies. It can be used to match a specific resource by name or
+// to match a set of resources by label.
+type PromotionPolicySelector struct {
+	// Name is the name of the resource to which this policy applies.
+	//
+	// It can be an exact name, a regex pattern (with prefix "regex:"), or a
+	// glob pattern (with prefix "glob:").
+	//
+	// NOTE: Using a specific exact name is the most secure option. Pattern
+	// matching via regex or glob can be exploited by users with permissions to
+	// match promotion policies that weren't intended to apply to their
+	// resources. For example, a user could create a resource with a name
+	// deliberately crafted to match the pattern, potentially bypassing intended
+	// promotion controls.
+	//
+	// +optional
+	Name string `json:"name,omitempty" protobuf:"bytes,1,opt,name=name"`
+
+	// LabelSelector is a selector that matches the resource to which this policy
+	// applies.
+	//
+	// NOTE: Using label selectors introduces security risks as users with
+	// appropriate permissions could create new resources with labels that match
+	// the selector, potentially enabling unauthorized auto-promotion.
+	// For sensitive environments, exact Name matching provides tighter control.
+	*metav1.LabelSelector `json:",inline" protobuf:"bytes,2,opt,name=labelSelector"`
+}
```

Comment:
Can we update the comments on the `Name` and `LabelSelector` fields to clarify matters of precedence? Does one take precedence over the other? Are the logically OR'ed?

Reply:
They are ANDed.